lib/ipfs/Ipfs.ts

```diff
@@ -0,0 +1,163 @@
+import * as crypto from 'crypto';
+import * as HttpStatus from 'http-status';
+import * as url from 'url';
+import base64url from 'base64url';
+import FetchResult from '../common/models/FetchResult';
+import FetchResultCode from '../common/enums/FetchResultCode';
+import ICas from '../core/interfaces/ICas';
+import IpfsErrorCode from '../ipfs/IpfsErrorCode';
+import nodeFetch from 'node-fetch';
+import ReadableStream from '../common/ReadableStream';
+import SharedErrorCode from '../common/SharedErrorCode';
+import SidetreeError from '../common/SidetreeError';
+import Timeout from './Util/Timeout';
+
+const multihashes = require('multihashes');
+
+/**
+ * Class that implements the `ICas` interface by communicating with IPFS.
+ */
+export default class Ipfs implements ICas {
+  private fetch = nodeFetch;
+
+  public constructor (public uri: string, private fetchTimeoutInSeconds: number) { }
```

Comment:
Any reason why this has to be `public`. Also, should we do any validation on it? I think we are letting the methods throw if the uri is invalid.

Reply:
No particular reason, will change it to private. What validation are you thinking of? Instance methods will throw if URI is invalid, which is okay, no?

Reply:
Yes that is fine. I was wondering whether we should stop initialization if the url is incorrect but then we're trying to move away from blocking the service start so it's ok.